packages/firebase_ml_vision/lib/src/barcode_detector.dart

```diff
@@ -4,18 +4,599 @@
+            ? null
+            : VisionBarcodePhone._(_data['phone']),
+        sms = _data['sms'] == null ? null : VisionBarcodeSMS._(_data['sms']),
+        url = _data['url'] == null
+            ? null
+            : VisionBarcodeURLBookmark._(_data['url']),
+        wifi =
+            _data['wifi'] == null ? null : VisionBarcodeWiFi._(_data['wifi']),
+        geoPoint = _data['geo_point'] == null
+            ? null
+            : VisionBarcodeGeoPoint._(_data['geo_point']),
+        contactInfo = _data['contact_info'] == null
+            ? null
+            : VisionBarcodeContactInfo._(_data['contact_info']),
+        calendarEvent = _data['calendar_event'] == null
+            ? null
+            : VisionBarcodeCalendarEvent._(_data['calendar_event']),
+        driverLicense = _data['driver_license'] == null
+            ? null
+            : VisionBarcodeDriverLicense._(_data['driver_license']);
+
+  /// Gets the bounding rectangle of the detected barcode.
+  ///
+  /// Returns null if the bounding rectangle can not be determined.
+  final Rectangle<int> boundingBox;
+
+  /// Returns barcode value as it was encoded in the barcode.
+  /// Structured values are not parsed, for example: 'MEBKM:TITLE:Google;URL://www.google.com;;'.
+  ///
+  /// Returns null if nothing found.
+  final String rawValue;
+
+  /// Returns barcode value in a user-friendly format.
+  /// May omit some of the information encoded in the barcode.
+  /// For example, if getRawValue() returns 'MEBKM:TITLE:Google;URL://www.google.com;;',
+  /// the display_value might be '//www.google.com'.
+  /// If valueFormat==TEXT, this field will be equal to getRawValue().
+  ///
+  /// This value may be multiline, for example, when line breaks are encoded into the original TEXT barcode value.
+  /// May include the supplement value.
+  ///
+  /// Returns null if nothing found.
+  final String displayValue;
+
+  /// Returns barcode format, for example [VisionBarcodeFormat.EAN13].
+  final VisionBarcodeFormat format;
+
+  /// Returns four corner points in clockwise direction starting with top-left.
+  ///
+  /// Due to the possible perspective distortions, this is not necessarily a rectangle.
+  final List<Point<int>> cornerPoints;
+
+  /// Returns format type of the barcode value.
+  ///
+  /// For example, [VisionBarcodeValueType.Text], [VisionBarcodeValueType.Product], [VisionBarcodeValueType.URL], etc.
+  ///
+  /// If the value structure cannot be parsed, TYPE_TEXT will be returned.
+  /// If the recognized structure type is not defined in your current version of SDK, TYPE_UNKNOWN will be returned.
+  ///
+  /// Note that the built-in parsers only recognize a few popular value structures.
+  /// For your specific use case, you might want to directly consume getRawValue()
+  /// and implement your own parsing logic.
+  final VisionBarcodeValueType valueType;
+
+  /// Gets parsed email details (set iff [valueType] is [VisionBarcodeValueType.Email].
```

Comment:
Missing closing parentheses at the end. You can remove `Gets` and just have `Parsed email details ...`